clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ElasticSearchServerGroupCachingAgent.groovy

```diff
@@ -0,0 +1,231 @@
+    return AwsProvider.PROVIDER_NAME;
+  }
+
+  @Override
+  public void run() {
+    Set<String> previousIndexes = new HashSet<>();
+
+    try {
+      JestResult result = jestClient.execute(
+        new GetAliases.Builder().build()
+      );
+
+      Map<String, Object> r = (Map<String, Object>) objectMapper.readValue(result.getJsonString(), Map.class);
+      for (String index : r.keySet()) {
+        if (index.startsWith("aws_server_groups")) {
+          previousIndexes.add(index);
+        }
+      }
+
+      System.out.println("Previous Indexes: " + previousIndexes);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    String indexName = "aws_server_groups_" + System.currentTimeMillis();
+
+    try {
+      jestClient.execute(
+        new CreateIndex.Builder(indexName).build()
+      );
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+
+    for (NetflixAmazonCredentials credentials : accounts) {
+      for (AmazonCredentials.AWSRegion region : credentials.getRegions()) {
+        AmazonAutoScaling amazonAutoScaling = amazonClientProvider.getAutoScaling(credentials, region.getName());
+
+        List<AutoScalingGroup> autoScalingGroups = amazonAutoScaling
+          .describeAutoScalingGroups()
+          .getAutoScalingGroups();
+        System.out.println("ASG: " + credentials.getName() + " - " + region.getName() + " - " + autoScalingGroups.size());
+
+        List<LaunchConfiguration> launchConfigurations = amazonAutoScaling
+          .describeLaunchConfigurations()
+          .getLaunchConfigurations();
+//        System.out.println("LC: " + credentials.getName() + " - " + region.getName() + " - " + launchConfigurations.size());
+
+        Map<String, LaunchConfiguration> launchConfigurationsByName = launchConfigurations.collectEntries {
+          [it.launchConfigurationName, it]
+        }
+
+        AmazonEC2 amazonEC2 = amazonClientProvider.getAmazonEC2(credentials, region.getName());
+        List<Instance> instances = new ArrayList<>();
+
+        for (Reservation reservation : amazonEC2.describeInstances().getReservations()) {
+//          System.out.println("I: " + credentials.getName() + " - " + region.getName() + " - " + reservation.getInstances().size());
+          instances.addAll(reservation.getInstances());
+        }
+
+        Map<String, Instance> instancesById = instances.collectEntries {
+          [it.instanceId, it]
+        }
+
+        Bulk.Builder builder = new Bulk.Builder().defaultIndex(indexName)
+
+        def serverGroups = autoScalingGroups.collect { AutoScalingGroup asg ->
+          def asgInstances = asg.instances.collect {
+            instancesById[it.instanceId]
+          }
+
+          def asgInstanceTypes = asgInstances*.instanceType as Set<String>
+
+          return [
+              id: "${credentials.accountId}:${region.name}:${asg.autoScalingGroupName.toLowerCase()}".toString(),
+              name: asg.autoScalingGroupName.toLowerCase(),
```

Comment:
Not sure how es querying works, but might be nice to split this into its parts to allow searching for specific stacks/details/applications

Reply:
Oh ya, certainly ... this was just super quick. 

We would actually need to index more data, parse names with frigga, etc. and have a proper es index template.

Funny enough, even just getting the data in probably offers very similar search capabilities (if not more) than the current redis search. There's only so much you can glob against the key patterns.